model-api/bpmn-model/src/main/java/org/camunda/bpm/model/bpmn/Bpmn.java

```diff
@@ -189,6 +357,10 @@ public static BpmnModelInstance createEmptyModel() {
   }
 
   public static ProcessBuilder createProcess() {
+    return createProcess(false);
+  }
+
+  public static ProcessBuilder createProcess(boolean skipDefaultHistoryTimeToLive) {
```

Comment:
❌ Do we really want to introduce this skip flag? 

We could instead make passing `#camundaHistoryTimeToLive(null)` work. Passing `null` to `#camundaHistoryTimeToLiveString` seems to work already.

From a user perspective, I find using the fluent builder API methods `#camundaHistoryTimeToLive` and `#camundaHistoryTimeToLiveString` much more intuitive than using a skip flag when starting the builder instance.

Reply:
❓ There is only one questionable case where initialising `historyTimeToLive` to the default value and then setting it to `null` does not work and it's not clear why.

This happens only in the case of `HistoryCleanupRemovalTimeTest#shouldCleanupProcessInstanceWithoutTTLWithConfigDefault`.

Reply:
Findings:

The two setters set the following value when called with `null`:
- `setCamundaHistoryTimeToLive`: sets `"null"` as HTTL
- `setCamundaHistoryTimeToLiveString`: sets `""` as HTTL

The tests had `enforceHistoryTimeToLive` turned off, so there is no exception thrown.

Suggested changes:

1. Pass `null` in [`ProcessImpl.setCamundaHistoryTimeToLive(Integer)`](https://github.com/camunda/camunda-bpm-platform/blob/master/model-api/bpmn-model/src/main/java/org/camunda/bpm/model/bpmn/impl/instance/ProcessImpl.java#L303), if parameter is `null`, instead of the `"null"` string.
2. Use `removeAttribute` in the [`ProcessImpl.setCamundaHistoryTimeToLiveString`](https://github.com/camunda/camunda-bpm-platform/blob/master/model-api/bpmn-model/src/main/java/org/camunda/bpm/model/bpmn/impl/instance/ProcessImpl.java#L313), this method needs to be exposed.
3. Rollback boolean skip flags.

Reply:
The problem was the handling of `null` value when using the `camundaHistoryTimeToLive` builder method after the initialisation of HTTL at `createExecutableProcess`.

After going through the problem with @danielkelemen, we decided to remove the `historyTimeToLive` attribute when setting `null` value through the model API. That solved the problem also of the failing test which had processes created with double setting of the field (first to default value and then to null)